compactor_scheduler/src/local_scheduler/catalog_commit.rs

```diff
@@ -37,13 +37,15 @@ impl Commit for CatalogCommit {
         upgrade: &[ParquetFile],
         create: &[ParquetFileParams],
         target_level: CompactionLevel,
-    ) -> Vec<ParquetFileId> {
-        assert!(!upgrade.is_empty() || (!delete.is_empty() && !create.is_empty()));
+    ) -> Result<Vec<ParquetFileId>, Error> {
+        if !(!upgrade.is_empty() || (!delete.is_empty() && !create.is_empty())) {
+            return Err(Error::BadRequest("commit must have files to upgrade, and/or a set of files to replace (delete and create)".into()));
+        }
```

Comment:
All these negatives and logical operators are making my head hurt :( I think this is saying:

- Either there must be files to upgrade, or 
- there must be both files to delete and create

is that right?

If so, I think this would be better as something like:

```suggestion
        if upgrade.is_empty() {
            match (delete.is_empty(), create.is_empty()) {
                (true, true) => return Err(Error::BadRequest("Commit was missing files to upgrade, files to delete, and files to create")),
                (true, false) => return Err(Error::BadRequest("Commit was missing files to upgrade and files to delete")),
                (false, true) => return Err(Error::BadRequest("Commit was missing files to upgrade and files to create")),
                _ => {},
            }
        }
```

What do you think?

Reply:
Used a match, but did it slightly different to make more explicit. See [here](https://github.com/influxdata/influxdb_iox/pull/8216/commits/e0081d7289e632fc66b4b97eb6516cf2c343c664) and tell me what you think, please!